src/test/java/org/zalando/jackson/datatype/money/MonetaryAmountDeserializerTest.java

```diff
@@ -1,147 +1,242 @@
+        final MonetaryAmount amount = unit.readValue(content, type);
 
         assertThat(amount, is(notNullValue()));
-        
+
         // we need a json node to get a TreeTraversingParser with codec of type ObjectReader
-        JsonNode ownerNode = unit.readTree("{ \"value\" : {\"amount\":29.95,\"currency\":\"EUR\",\"formatted\":\"30.00 EUR\"} }");
-        
-        
+        final JsonNode ownerNode =
+                unit.readTree("{\"value\":{\"amount\":29.95,\"currency\":\"EUR\",\"formatted\":\"30.00EUR\"}}");
+
         final Owner owner = new Owner();
         owner.setValue(amount);
-        
+
         // try to update
-        Owner result = unit.readerForUpdating(owner).readValue(ownerNode);
+        final Owner result = unit.readerForUpdating(owner).readValue(ownerNode);
         assertThat(result, is(notNullValue()));
         assertThat(result.getValue(), is(amount));
     }
-    
+
     private static class Owner {
-        
+
         private MonetaryAmount value;
 
         public MonetaryAmount getValue() {
             return value;
         }
 
-        public void setValue(MonetaryAmount value) {
+        public void setValue(final MonetaryAmount value) {
             this.value = value;
         }
+
+    }
+
+    @Test
+    public void shouldFailToDeserializeWithoutAmount() throws IOException {
+        final ObjectMapper unit = unit();
+
+        final String content = "{\"currency\":\"EUR\"}";
+
+        exception.expect(JsonProcessingException.class);
+        exception.expectMessage("Missing property: 'amount'");
+
+        unit.readValue(content, type);
+    }
+
+    @Test
+    public void shouldFailToDeserializeWithoutCurrency() throws IOException {
+        final ObjectMapper unit = unit();
+
+        final String content = "{\"amount\":29.95}";
+
+        exception.expect(JsonProcessingException.class);
+        exception.expectMessage("Missing property: 'currency'");
+
+        unit.readValue(content, type);
+    }
+
+    @Test
+    public void shouldNotFailToDeserializeWithFormatted() throws IOException {
+        final ObjectMapper unit = unit();
+
+        final String content = "{\"amount\":29.95,\"currency\":\"EUR\",\"formatted\":\"30.00 EUR\"}";
+
+        unit.readValue(content, type);
+    }
+
+    @Test
+    public void shouldFailToDeserializeWithAdditionalProperties() throws IOException {
+        final ObjectMapper unit = unit();
+
+        final String content = "{\"amount\":29.95,\"currency\":\"EUR\",\"version\":\"1\"}";
+
+        exception.expect(UnrecognizedPropertyException.class);
+        exception.expectMessage(startsWith(
+                "Unrecognized field \"version\" (class javax.money.MonetaryAmount), " +
+                        "not marked as ignorable (3 known properties: \"amount\", \"currency\", \"formatted\"])"));
+
+        unit.readValue(content, type);
+    }
+
+    @Test
+    public void shouldNotFailToDeserializeWithAdditionalProperties() throws IOException {
```

Comment:
I have a complex object in that test already. In fact I used this very test to verify the need for `skipChildren`.